dts/bindings/dac/atmel,samd5x-dac.yaml

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Atmel SAMD5x/E5x family DAC
+  
+  To enable the DAC peripheral, it must be enabled in the board dts, or in 
+  subsequent overlay file. In addition, each channel must be enabled and 
+  configured via the array properties 'channel-one-config' and 
+  'channel-two-config' which details are layout in the properties description.
+
+  &dac0 {
+    status = "okay";
+    reference = "intref";
+    channel-one-config = <0 0 1 0 0 2 0 1>;
+    channel-two-config = <0 0 1 0 0 2 0 1>;
+  };
+
+compatible: "atmel,samd5x-dac"
+
+include:
+  - name: dac-controller.yaml
+  - name: pinctrl-device.yaml
+  - name: atmel,assigned-clocks.yaml
+
+properties:
+  reg:
+    required: true
+
+  clocks:
+    required: true
+
+  clock-names:
+    required: true
+
+  atmel,assigned-clocks:
+    required: true
+
+  atmel,assigned-clock-names:
+    required: true
+
+  reference:
+    type: string
+    description: Reference voltage source
+    enum:
+      - "vrefau"
+      - "vddana"
+      - "vrefab"
+      - "intref"
+
+  differential-mode:
+    type: boolean
+    required: false
+    description: Enable differential mode (SAMD5x only)
+    
+  channel-one-config:
+    type: array
+    required: false
+    description: | 
+      Per-channel configuration (SAMD5x/E5x), set up in the following order
+      # - Oversampling ratio (0-x1, 1-x2, 2-x4, 3-x8, 4-x16, 5-x32)
+      # - Dither mode (0 disable, 1 enable)
+      # - Refresh period (0 disable, if (n > 0) T_refresh = n*30us)
+      # - Run in standby (0 disable, 1 enable)
+      # - External filter (0 disable, 1 enable)
+      # - Current control (0, 1, 2)
+      # - Left adjusted (0 right-adjusted, 1 left-adjusted)
+      # - Enable channel (0 disable, 1 enable)
+    default: [0, 0, 0, 0, 0, 0, 0, 0]
+  
+  channel-two-config:
+    type: array
+    required: false
+    description: |
+      Per-channel configuration (SAMD5x/E5x), set up in the following order
+      # - Oversampling ratio (0-x1, 1-x2, 2-x4, 3-x8, 4-x16, 5-x32)
+      # - Dither mode (0 disable, 1 enable)
+      # - Refresh period (0 disable, if (n > 0) T_refresh = n*30us)
+      # - Run in standby (0 disable, 1 enable)
+      # - External filter (0 disable, 1 enable)
+      # - Current control (0, 1, 2)
+      # - Left adjusted (0 right-adjusted, 1 left-adjusted)
+      # - Enable channel (0 disable, 1 enable)
+    default: [0, 0, 0, 0, 0, 0, 0, 0]
```

Comment:
Let's use child nodes with named properties (one child for every DAC channel) instead of the array. The array parameters will look like lots of magic numbers in the board devicetree file.